itou/job_applications/models.py

```diff
@@ -208,7 +208,11 @@ def eligible_as_employee_record(self, siae):
             # Exclude flagged approvals (batch creation or import of approvals)
             .exclude(approval__create_employee_record=False)
             # Exclude existing employee records with the same PASS IAE and to_siae.asp_id (considered as dups by ASP)
-            .exclude(employee_record__asp_id=siae.asp_id, employee_record__approval_number=F("approval__number"))
+            .exclude(
+                ~Q(to_siae=siae),
```

Comment:
J'ai l'impression que c'est redondant avec le `filter(to_siae=siae)` un peu plus bas, non ? On se limitait déjà aux job applications de cette siae. Excluer ici les job applications qui ne sont pas de cette siae est redondant si j'ai bien suivi.

Reply:
Mmm au temps pour moi il y a bien un impact. Ton changement fait qu'on exclut moins de JA qu'avant. Celles qu'on excluait avant mais plus après, ce sont les JA de cette SIAE (qui ont le même PASS IAE et ASP ID). Mais tout de même la combinaison de ce `exclude` et de ce `filter` plus bas me fait des noeuds au cerveau.

Peut-être qu'on pourrait les remplacer par un `filter(to_siae=siae AND (NOT the same asp id OR NOT the same PASS IAE))`.

Reply:
Cf https://itou-inclusion.slack.com/archives/C01181Y04LT/p1638204451311200

Ma meilleure proposition serait pour une meilleure compréhension de refactorer ça en une méthode `siae.eligible_job_applications()` comme ça on a naturellement en entrée que les JA de la bonne SIAE, et moins de noeuds au cerveau.

Reply:
Effectivement, j'ai essayé de faire moins de nœuds au cerveau avec une `subquery` et en documentant.
Merci de ton retour !